src/modules/list.js

```diff
@@ -0,0 +1,62 @@
+export default class List {
+  constructor(ItemType, listName, storageName = 'list') {
+    this.ItemType = ItemType;
+    this.storageName = storageName;
+    this.listName = listName;
+    this.list = {};
+    this.#retrieveStorage();
+  }
+
+  #addItem(item) {
+    this.list[item.index] = item;
+  }
+
+  #removeItem(item) {
+    delete this.list[item.index];
+  }
+
+  #updateStorage() {
+    localStorage.setItem(this.storageName, JSON.stringify(this.list));
+  }
+
+  updateStorage() {
+    this.#updateStorage();
+  }
+
+  #retrieveStorage() {
+    if (localStorage.getItem(this.storageName) === null) {
+      this.#updateStorage();
+    } else {
+      const tempList = JSON.parse(localStorage.getItem(this.storageName));
+      Object.values(tempList).forEach((item) => {
+        this.list[item.index] = new this.ItemType(item.description, item.completed, item.index);
+      });
+    }
+  }
+
+  addItem(item) {
+    this.#addItem(item);
+    this.#updateStorage();
+  }
+
+  removeItem(item) {
+    this.#removeItem(item);
+    this.#updateStorage();
+  }
```

Comment:
- [ ] I can see you implemented the `removeItem` function but you didn't add a button or link for each task to make you able to remove every task indvidually so please fix that by adding a button and event listener for removing a single task.

Reply:
As I mentioned about this in my PR there is no specific requirement in this task for index list items. Kindly have a look at this image. I think it's unnecessary to ask the same thing again & again also I explained everything.

[![Minimalist example](https://camo.githubusercontent.com/d04bee7a5b8f62cc6655dbec65c4d14583e4d78d0c434ddc070f6881bb9568fc/68747470733a2f2f696d672e796f75747562652e636f6d2f76692f416355642d5f596a6a71672f302e6a7067)](https://www.youtube.com/watch?v=AcUd-_Yjjqg)

Reply:
Hi @Fatima-hub333 
It's a project requirement to implement remove single-task functionality and you can see the [design ](https://web.archive.org/web/20180320194056/http://www.getminimalist.com:80/) given for this project when you click on the task to edit it the delete button for single task appeared like the image below.

![image](https://user-images.githubusercontent.com/68796232/166930280-5ad7ea0d-32c6-42fb-a5ff-fc92dcea4a90.png)